packages/core/src/hosted-form/hosted-form-vaulting.ts

```diff
@@ -0,0 +1,42 @@
+export interface HostedFormVaultingData {
+    currencyCode: string;
+    paymentsUrl: string;
+    providerId: string;
+    shopperId: string;
+    storeHash: string;
+    vaultToken: string;
+}
+
+export interface HostedFormVaultingPaymentInstrument {
+    type: string;
+    cardholder_name: string;
+    number: string;
+    expiry_month: number;
+    expiry_year: number;
+    verification_value: string;
+}
+
+export interface HostedFormVaultingBillingAddress {
+    address1: string;
+    address2?: string;
+    city: string;
+    postal_code: string;
+    country_code: string;
+    company?: string;
+    first_name: string;
+    last_name: string;
+    email: string;
+    phone?: string;
+    state_or_province_code?: string;
+}
+
+export interface HostedFormVaultingInstrumentForm {
+    billingAddress: HostedFormVaultingBillingAddress;
```

Comment:
Is there a reason we have a mix of camel case and snake case?

Reply:
@animesh1987 the reason was to use common practice from the checkout SDK by naming in the camel case, but endpoint expects to get fields in the snake case. 
Yeah, I agree with you would be better to rename everything to the snake case I believe.

Reply:
Yeah, great idea!